client_test.go

```diff
@@ -56,7 +50,44 @@ func TestMakeRequest(t *testing.T) {
 		req := client.newRequest("GET", "/v2/foobar")
 		resp, err := client.doRequest(req)
 		So(err, ShouldBeNil)
-		So(client.GetToken(), ShouldEqual, "bearer foobar")
 		So(resp, ShouldNotBeNil)
 	})
 }
+
+func TestTokenRefresh(t *testing.T) {
+	gomega.RegisterTestingT(t)
+	Convey("Test making request", t, func() {
+		server := fakeUAAServer()
+		c := &Config{
+			ApiAddress:   server.URL,
+			LoginAddress: server.URL,
+			Username:     "foo",
+			Password:     "bar",
+		}
+		client := NewClient(c)
+		gomega.Consistently(client.GetToken()).Should(gomega.Equal("bearer foobar2"))
+		gomega.Eventually(client.GetToken(), "3s").Should(gomega.Equal("bearer foobar3"))
```

Comment:
"3s" is a timeout, so within 3s `client.GetToken()` should return `"bearer foobar3"` ([docs](http://onsi.github.io/gomega/#eventually)). So it is not actually waiting 3s.